src/tooltip/Tooltip2.tsx

```diff
@@ -0,0 +1,322 @@
+import React, {
+  Fragment,
+  useState,
+  useEffect,
+  useRef,
+  useCallback,
+} from 'react';
+import {
+  TouchableOpacity,
+  Modal,
+  View,
+  StatusBar,
+  I18nManager,
+  ViewStyle,
+  FlexStyle,
+  StyleProp,
+  StyleSheet,
+  ColorValue,
+} from 'react-native';
+import { act } from 'react-test-renderer';
+import { withTheme } from '../config';
+import { ScreenWidth, ScreenHeight, isIOS } from '../helpers';
+import Triangle from './Triangle';
+import getTooltipCoordinate, {
+  getElementVisibleWidth,
+} from './getTooltipCoordinate';
+
+export type TooltipProps = {
+  withPointer?: boolean;
+  popover?: React.ReactElement<{}>;
+  toggleOnPress?: boolean;
+  toggleAction?: 'onPress' | 'onLongPress';
+  height?: FlexStyle['height'];
+  width?: FlexStyle['width'];
+  containerStyle?: StyleProp<ViewStyle>;
+  pointerColor?: ColorValue;
+  onClose?(): void;
+  onOpen?(): void;
+  overlayColor?: ColorValue;
+  withOverlay?: boolean;
+  backgroundColor?: ColorValue;
+  highlightColor?: ColorValue;
+  skipAndroidStatusBar?: boolean;
+  ModalComponent?: typeof React.Component;
+  closeOnlyOnBackdropPress?: boolean;
+} & typeof defaultProps;
+
+const defaultProps = {
+  withOverlay: true,
+  overlayColor: 'rgba(250, 250, 250, 0.70)',
+  highlightColor: 'transparent',
+  withPointer: true,
+  toggleOnPress: true,
+  toggleAction: 'onPress',
+  height: 40,
+  width: 150,
+  containerStyle: {},
+  backgroundColor: '#617080',
+  onClose: () => {},
+  onOpen: () => {},
+  skipAndroidStatusBar: false,
+  ModalComponent: Modal,
+  closeOnlyOnBackdropPress: false,
+};
+
+const Tooltip: React.FunctionComponent<TooltipProps> = (props) => {
+  const renderedElement = useRef(null);
+  const [isVisible, setIsVisible] = useState(false);
+  const [yOffset, setYOffset] = useState(0);
+  const [xOffset, setXOffset] = useState(0);
+  const [elementWidth, setElementWidth] = useState(0);
+  const [elementHeight, setElementHeight] = useState(0);
+
+  const {
+    withOverlay = true,
+    overlayColor = 'rgba(250, 250, 250, 0.70)',
+    highlightColor = 'transparent',
+    withPointer = true,
+    toggleOnPress = true,
+    toggleAction = 'onPress',
+    height = 40,
+    width = 150,
+    backgroundColor = '#617080',
+    onClose = () => {},
+    onOpen = () => {},
+    skipAndroidStatusBar = false,
+    ModalComponent = Modal,
+    closeOnlyOnBackdropPress = false,
+    pointerColor,
+  } = props;
+
+  const toggleTooltip = () => {
+    getElementPosition();
```

Comment:
Why is this called here?

Reply:
That part is actually useful for ensuring absolute position one last time before showing e.g. in a changing layout such as flat list however it's a bit unnecessary before hiding.

Reply:
Actually now that I think about it if we consider this usage as safe and not overkill then the one called at mount becomes redundant so basically tooltip will hold render until next coordinates are available before each show. This can be easily achieved by clearing coordinates before hide and set before show.

Or alternatively this kind of behavior can be actually configurable in case of static layouts, either dynamic positioning which calculates before each trigger or static; use always same the one calculated during mount.